src/pages/movie-description.jsx

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import MoviesList from "../components/movies-list";
+import MovieDetails from "../components/movie-details";
+import { MovieService } from "../movie-service";
+import { ErrorMessage } from "../shared-components/error-message";
+
+
+export class MovieDescription extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            currentMovie: {},
+            similarGenreFilms: [],
+            error: null
+        };
+        this.movieService = new MovieService();
+    }
+
+    componentDidMount() {
+        this.movieService.getMovie(this.props.match.params.id)
+            .then(movie => {
+                this.setState({ currentMovie: movie })
+                this.movieService.searchMovie('genres', movie.genres[0])
+                    .then(similarMovies => this.setState({ similarGenreFilms: similarMovies })
+                        .catch(error => this.setState({ error })))
+            }).catch(error => this.setState({ error }));
```

Comment:
one more thing to note. `fetch` API returns successfully resolved promise even if server responded with 4** or 5** status. If you do not throw an error in your `movieService`, your catch callback will never fire.

So, if you wish to handle 4** and 5** responses as errors, you will have to do that is your service. Something like that:

```javascript
fetch(/* params */)
  .then(response => {
    if (response.ok) {
      throw new Error(`Request failed with status ${response.status})`
    }
    return response.json()
  })
```